packages/web-app/src/components/appli/Entry/Snapshots/index.jsx

```diff
@@ -0,0 +1,188 @@
+import React, { useState, useEffect } from 'react';
+import { useIntl } from 'react-intl';
+import { useParams, useLocation } from 'react-router-dom';
+import { useDispatch, useSelector } from 'react-redux';
+import {
+  Accordion,
+  AccordionSummary,
+  AccordionDetails,
+  AccordionActions,
+  Typography,
+  Box
+} from '@material-ui/core';
+import { Skeleton } from '@material-ui/lab';
+import ExpandMore from '@material-ui/icons/ExpandMore';
+import { pathOr } from 'ramda';
+
+import {
+  CommentSnapshots,
+  DocumentSnapshots,
+  EntranceCaveSnapshots,
+  EntranceNetworkSnapshots,
+  GenericSnapshots,
+  RiggingSnapshots
+} from './component/SnapshotComponents';
+import Alert from '../../../common/Alert';
+import Contribution from '../../../common/Contribution/Contribution';
+import { fetchSnapshot } from '../../../../actions/Snapshot/GetSnapshots';
+import REDUCER_STATUS from '../../../../reducers/ReducerStatus';
+import ScrollableContent from '../../../common/Layouts/Fixed/ScrollableContent';
+import getAuthor from '../../../../util/getAuthor';
+import SensitiveCaveWarning from '../SensitiveCaveWarning';
+
+function renderComponent(type, data, isNetwork) {
+  switch (type) {
+    case 'riggings':
+      return <RiggingSnapshots rigging={data} key={data.id} />;
+    case 'entrances':
+      return isNetwork ? (
+        <EntranceNetworkSnapshots entrance={data} key={data.id} />
+      ) : (
+        <EntranceCaveSnapshots entrance={data} key={data.id} />
+      );
+    case 'documents':
+      return <DocumentSnapshots document={data} key={data.id} />;
+    case 'comments':
+      return <CommentSnapshots comment={data} key={data.id} />;
+    default:
+      return <GenericSnapshots data={data} key={data.id} />;
+  }
+}
+
+const SnapshotPage = () => {
+  const { formatMessage } = useIntl();
+  const dispatch = useDispatch();
+  const location = useLocation();
+  const queryParameters = new URLSearchParams(location.search);
+  const isNetwork = queryParameters.get('isNetwork') === 'true';
+  const [actualTItem, setActualTItem] = useState({});
+
+  useEffect(() => {
+    const json = localStorage.getItem('t_item');
```

Comment:
Not sure I understand why we need localStorage. Can I get some context please?

Reply:
In the snapshots page, we need to show the current object. 
So to avoid multiple API call, we use the local storage to store the current object and get it back when we arrive on the new component. I delete it right after getting it to avoid unecessary stored items.  

Why the local storage ? 
When we click on the button to open a snapshots page, we need to open a new tab. So we cannot use the things provided by `Link` because it's only related to a tab.